src/update_agent/actor.rs

```diff
@@ -131,23 +133,29 @@ impl Handler<RefreshTick> for UpdateAgent {
                 UpdateAgentState::EndState => (),
             };
 
+            // Update state_changed timestamp if necessary.
+            if discriminant(&prev_state) != discriminant(&*agent_state_guard) {
+                let cur_timestamp = chrono::Utc::now();
+                // This mutable borrow will not panic because we are still holding
```

Comment:
Self-note: we may need to revisit this statement as soon as we add a consumer for this field, which may or may not result in other consumers having an immutable borrow when we reach here. We'll likely need to replace the `RefCell` usage at that point, but we can cross that bridge once we get there.

Reply:
Ah right, I don't think my comment there is true once there are consumers. I completely forgot that consumers need to borrow the field for reading as well.

Reply:
Ah right, I don't think my comment there is true once there are consumers. I completely forgot that consumers need to borrow the field for reading as well.